applications/CompressiblePotentialFlowApplication/custom_conditions/potential_wall_condition.h

```diff
@@ -202,6 +206,125 @@ class PotentialWallCondition : public Condition
         return pNewCondition;
     }
 
+    /// Find the condition's parent element.
+	void Initialize() override
+	{
+        KRATOS_TRY;
+
+        if(this->Id()==1)
+        {
+            std::cout << "Condition = " << this->Id()  << std::endl;
+            std::cout << "mInitializeWasPerformed = " << mInitializeWasPerformed  << std::endl;
+        }
+        
+        //std::cout << "Condition = " << this->Id()  << std::endl;
+
+		const array_1d<double,3>& rNormal = this->GetValue(NORMAL);
+		if (norm_2(rNormal) == 0.0)
+		  {
+		    std::cout << "error on condition -> " << this->Id() << std::endl;
+		    KRATOS_THROW_ERROR(std::logic_error, "NORMAL must be calculated before using this condition","");
```

Comment:
Not a blocker, but try to use the newer style `KRATOS_ERROR_IF(...) << "Error on condition ..." ...`

Reply:
and do not compare doubles with ==